src/views/WorkoutView.tsx

```diff
@@ -0,0 +1,268 @@
+                    currentPhase={currentPhase as TimerPhase}
+                    description={currentTimer.description}
+                    running={running}
+                    completed={isCompleted}
+                    isFirstTimer={isFirstTimer}
+                    hasStarted={hasStarted}
+                    handleStart={runWorkout}
+                    handleReset={handleReset}
+                    handleFastForward={handleFastForward}
+                />
+                {!((isFirstTimer && !hasStarted) || isCompleted) && <div className="text-sm text-gray-400 mt-2">{displayTimerDetails(currentTimer)}</div>}
+            </Panel>
+        );
+    };
+
+    // Render queue of timers
+    return (
+        <div className="flex flex-col items-center w-full max-w-2xl px-4 mx-auto">
+            <h1>Workout</h1>
+
+            {renderCurrentTimer()}
+
+            <div className="w-full max-w-md">
+                {timers.map((timer, index) => (
+                    <div
+                        key={index}
+                        className={`
+            flex
+            justify-between
+            items-center
+            mb-2 
+            p-2 
+            rounded-lg
+            bg-slate-900/50
+            ${timer.state === 'completed' ? 'line-through text-gray-600' : ''}
+            ${hasStarted && index === currentTimerIndex ? 'text-lime-200' : ''}
+          `}
+                    >
+                        <div className="flex flex-col">
+                            <Button
+                                onClick={() => handleMoveUp(index)}
+                                disabled={
+                                    (hasStarted && index <= currentTimerIndex) ||
+                                    timer.state === 'completed' ||
+                                    index === 0 ||
+                                    (index > 0 && (timers[index - 1].state === 'completed' || timers[index - 1].state === 'running'))
+                                }
+                            >
+                                ↑
+                            </Button>
+                            <Button onClick={() => handleMoveDown(index)} disabled={(hasStarted && index <= currentTimerIndex) || timer.state === 'completed' || index === timers.length - 1}>
+                                ↓
+                            </Button>
+                        </div>
+                        <div className="flex flex-col">
+                            <div className="flex-grow text-center">{displayTimerDetails(timer)}</div>
+                            {timer.description && <div className="text-sm text-gray-500 text-center mt-1">{timer.description}</div>}
+                        </div>
+                        <div className="flex flex-col gap-2 ml-4">
+                            <Button onClick={() => editTimer(index)} disabled={(hasStarted && index <= currentTimerIndex) || timer.state === 'completed'}>
```

Comment:
if i edit a timer after a workout run and then run, it does not store a record in my workout history